### PR TITLE
Closes #1366 Allow upgrade of smart_date and fix demo menu links migration.

### LINKED ISSUE #1366: Latest Smart Date module release causes PHPunit test failures
## Problem/Motivation
Some of our PHPunit tests fail with the latest release of the Drupal contrib Smart Date module (3.5.0).

(See the ProboCI checks on #1272)

The failures seem to be associated with the [SmartDateCreateRecurrenceOnField](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_event/src/Plugin/migrate/process/SmartDateCreateRecurrenceOnField.php) migrate process plugin included in the Quicsktart Event module.

### Describe the bug
```
Runtime:       PHP 7.4.26

Configuration: /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/phpunit.xml.dist
EE                                                                  2 / 2 (100%)

Time: 6.48 minutes, Memory: 10.00 MB

There were 2 errors:

1) Drupal\Tests\az_demo\Functional\AZDemoContentTest::testUtilityLinks
array_flip(): Can only flip STRING and INTEGER values!

/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php:261
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php:245
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:281
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:948
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:296
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:232
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:165
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_event/src/Plugin/migrate/process/SmartDateCreateRecurrenceOnField.php:45
/src/az-quickstart-scaffolding/web/core/modules/migrate/src/MigrateExecutable.php:399
/src/az-quickstart-scaffolding/web/core/modules/migrate/src/MigrateExecutable.php:205
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_demo/az_demo.install:23
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Extension/ModuleHandler.php:392
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Extension/ModuleInstaller.php:343
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php:83
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Test/FunctionalTestSetupTrait.php:474
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:562
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:381
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_demo/tests/src/Functional/AZDemoContentTest.php:47
/src/az-quickstart-scaffolding/vendor/phpunit/phpunit/src/Framework/TestResult.php:703

2) Drupal\Tests\az_demo\Functional\AZDemoContentTest::testTitle
array_flip(): Can only flip STRING and INTEGER values!

/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php:261
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php:245
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:281
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:948
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:296
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:232
/src/az-quickstart-scaffolding/web/modules/contrib/smart_date/modules/smart_date_recur/src/Entity/SmartDateRule.php:165
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_event/src/Plugin/migrate/process/SmartDateCreateRecurrenceOnField.php:45
/src/az-quickstart-scaffolding/web/core/modules/migrate/src/MigrateExecutable.php:399
/src/az-quickstart-scaffolding/web/core/modules/migrate/src/MigrateExecutable.php:205
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_demo/az_demo.install:23
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Extension/ModuleHandler.php:392
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Extension/ModuleInstaller.php:343
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php:83
/src/az-quickstart-scaffolding/web/core/lib/Drupal/Core/Test/FunctionalTestSetupTrait.php:474
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:562
/src/az-quickstart-scaffolding/web/core/tests/Drupal/Tests/BrowserTestBase.php:381
/src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/custom/az_demo/tests/src/Functional/AZDemoContentTest.php:47
/src/az-quickstart-scaffolding/vendor/phpunit/phpunit/src/Framework/TestResult.php:703

ERRORS!
Tests: 2, Assertions: 0, Errors: 2.
```
## Proposed resolution
Identify and fix cause of test failures so we can upgrade Smart Date.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Smart date module updates introduced some new features for timezones that interfere with our demo content migrations.

Additionally, `az_demo_menu_links` migration errors because the bundle is being set to null because there is no `menu_link_content` source field.  So we should change that to use the default_value process plugin instead.

## Related issues
Fixes #1366 

## How to test
View [this node on probo](https://d092ca03-0bb4-4339-b539-84f6e7cf1071.probo.build/events/weekend-canyon-running-training), see if recurring event info still gets imported correctly.

## Types of changes
Updated smart_date to 3.5.1 from 3.4.3
Added patch fixing new issue from 3.5.0
Remove unneeded code from data json file for smart date migration.
Fix menu_link_content migration by setting the bundle on migrated menu_link_content entities.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [x] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_demo/migrations/az_demo_menu_links.yml [hunk 1]
```diff
@@ -51,7 +51,9 @@ source:
     link_id:
       type: integer
 process:
-  bundle: menu_link_content
+  bundle:
```
Comment (trackleft) [inline modules/custom/az_demo/migrations/az_demo_menu_links.yml:54]: There is no menu_link_content source field

File: composer.json [hunk 2]
```diff
@@ -154,6 +154,9 @@
             "drupal/media_migration": {
                 "Catch migrate RequirementsExceptions (3268480)": "https://www.drupal.org/files/issues/2022-03-08/3268480-2.patch"
             },
+            "drupal/smart_date": {
+                "PHP error when creating smart_date_rule entities that are not attached to entities (3274881)": "https://www.drupal.org/files/issues/2022-04-12/issue-3274881-6-mr.patch",
```
Comment (joeparsons) [inline composer.json:158]:
```suggestion
                "PHP error when creating smart_date_rule entities that are not attached to entities (3274881)": "https://www.drupal.org/files/issues/2022-04-12/issue-3274881-6-mr.patch"
```

File: modules/custom/az_demo/migrations/az_demo_menu_links.yml (hunk 1 above)
Comment (trackleft) [reply modules/custom/az_demo/migrations/az_demo_menu_links.yml:54]: Ok updated the PR description

**Comment by bberndt-uaz** [comment]
I'm not sure if I'm missing something, but should the times on the [Canyon Running test page](https://d092ca03-0bb4-4339-b539-84f6e7cf1071.probo.build/events/weekend-canyon-running-training) be from 4:30 am to 2:00 pm? Currently, the events are showing from 9:30 p.m. to 7 a.m. On another test environment I have with the demo content enabled, I see 4:30 a.m. to 2 p.m.

**Review by joeparsons — APPROVED** [review]
This looks good to me.

**Review by bberndt-uaz — APPROVED** [review]

**Pull request merged**